flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/consumer/RemoteInputChannel.java

```diff
@@ -313,6 +320,49 @@ public int getNumberOfAvailableBuffers() {
 		}
 	}
 
+	/**
+	 * Buffer pool notifies this channel of an available floating buffer. If the channel is released or not
+	 * needing extra buffers currently, the buffer should be recycled to buffer pool. Otherwise, the buffer
+	 * will be added into the available queue and the unannounced credit is increased by one.
```

Comment:
`The Buffer pool notifies this channel of an available floating buffer. If the channel is released or currently does not need extra buffers, the buffer should be recycled to the buffer pool. Otherwise, the buffer will be added into the <tt>availableBuffers</tt> queue and the unannounced credit is increased by one.`